backend/src/Database.hs

```diff
@@ -111,20 +116,83 @@ toFilterExpression ::
   SqlExpr (Value Bool)
 toFilterExpression report spec = foldr ((&&.) . fromMaybe (val True)) (val True) filterList
   where
-    toFilter entity field values = ((entity ^. field) `in_`) . valList <$> values
-    filterList = [playerFilter, pairingFilter, winnerFilter, loserFilter, leagueFilter]
-    playerFilter = liftA2 (||.) (toFilter report GameReportWinnerId spec.players) (toFilter report GameReportLoserId spec.players)
+    toInequalityFilter ::
+      (PersistEntity e) =>
+      SqlExpr (Entity e) -> EntityField e Int -> InequalityFilter -> SqlExpr (Value Bool)
+    toInequalityFilter entity field = \case
+      InequalityFilter LT a -> (entity ^. field) <. val a
+      InequalityFilter GT a -> (entity ^. field) >. val a
+      InequalityFilter EQ a -> (entity ^. field) ==. val a
+    toMaybeInequalityFilter ::
+      (PersistEntity e) =>
+      SqlExpr (Entity e) -> EntityField e (Maybe Int) -> Maybe InequalityFilter -> SqlExpr (Value Bool)
+    toMaybeInequalityFilter entity field = \case
+      Nothing -> isNothing_ (entity ^. field)
+      Just (InequalityFilter LT a) -> (entity ^. field) <. (just . val $ a)
+      Just (InequalityFilter GT a) -> (entity ^. field) >. (just . val $ a)
+      Just (InequalityFilter EQ a) -> (entity ^. field) ==. (just . val $ a)
+    toListFilter ::
```

Comment:
This is what used to just be called `toFilter`

Note that the type of the function has changed - it no longer expects values wrapped in a functor, but rather just the values themselves. This whole function can then be fmapped inside whatever functor it needs to be, when it is called. That seemed cleaner to me.